src/java.base/share/classes/java/net/InterfaceAddress.java

```diff
@@ -101,15 +101,10 @@ public short getNetworkPrefixLength() {
      * @see     java.net.InterfaceAddress#hashCode()
      */
     public boolean equals(Object obj) {
-        if (obj instanceof InterfaceAddress) {
-            InterfaceAddress cmp = (InterfaceAddress) obj;
-
-            if (Objects.equals(address, cmp.address) &&
-                Objects.equals(broadcast, cmp.broadcast) &&
```

Comment:
Apologies Patrick, building on Michael's suggestion, then you'd end up at:
 
     public boolean equals(Object obj) {
          return obj instanceof InterfaceAddress cmp
                   && Objects.equals(address, cmp.address)
                   && Objects.equals(broadcast, cmp.broadcast)
                   && maskLength == cmp.maskLength;
      }